drivers/ondisk_test.go

```diff
@@ -0,0 +1,149 @@
+package drivers
+
+import (
+	"os"
+	"path"
+	"testing"
+)
+
+func TestNewOnDiskStorageController(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+	if sc.FSPath != "test/controlla" {
+		t.Fatal("Storage controller created in wrond dir ", sc.FSPath)
+	}
+}
+
+func TestSCConnect(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+
+	err := sc.Connect()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+func TestSCDisconnect(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+
+	err := sc.Disconnect()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+func TestSCMount(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+	/*
+		mountedPath, err := sc.Mount("")
+
+		if err == nil {
+			t.Fatal("A valid mountedPath name should be provided to storage controller")
+		}
+	*/
+
+	mountedPath, err := sc.Mount("formulavol1")
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if mountedPath != "test/controlla/formulavol1" {
+		t.Fatal("Did not receive expected path, instead got ", mountedPath)
+	}
+
+	mountedPath, err = sc.Mount("formulavol1")
+
+	if err != nil {
+		t.Fatal(err, " strange, should have been able to mount the same vol")
+	}
+
+	if mountedPath != "test/controlla/formulavol1" {
+		t.Fatal("If I am mounting the same volume, the same path should be kept ", mountedPath)
+	}
+
+	os.Rename(mountedPath, path.Join(path.Dir(mountedPath), path.Base(mountedPath)+".unmounted"))
```

Comment:
I was trying to test mount without unmount. I feel like this coupling works as we don't expect the implementation to change.